realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -1048,6 +1081,151 @@ void BaseRealSenseNode::publishStaticTransforms()
     }
 }
 
+/* Modified for Brain */
+static void output_unit_vector(unsigned int height, unsigned int width, float unitVectors[FRAME_MAX_HEIGHT][FRAME_MAX_WIDTH * 3])
+{
+	unsigned int i, j;
+	FILE *fp; /* output to .obj file with a certain format to be shown by some tool */
```

Comment:
Careful using raw pointers. Consider using [file streams](http://www.cplusplus.com/doc/tutorial/files/) (_ifstream_ for input, _ofstream_ for output).